plugins/HostLinux.cpp

```diff
@@ -32,112 +32,127 @@ bool HostLinux::peek(const procptr_t address, void *dst, const size_t size) cons
 	return (ret != -1 && static_cast< size_t >(ret) == in.iov_len);
 }
 
-procptr_t HostLinux::module(const std::string &module) const {
-	std::stringstream ss;
-	ss << "/proc/";
-	ss << m_pid;
-	ss << "/maps";
-	const auto maps = readFile(ss.str());
+Modules HostLinux::modules() const {
+	std::ostringstream path;
+	path << "/proc/";
+	path << m_pid;
+	path << "/maps";
+	const auto maps = readFile(path.str());
 
 	if (maps.size() == 0) {
-		return 0;
+		return {};
 	}
 
-	std::stringstream ssPath(maps);
-	while (ssPath.good()) {
-		std::string baseaddr;
-
-		int ch;
-		while (1) {
-			ch = ssPath.get();
-			if (ch == '-') {
-				break;
-			} else if (ch == EOF) {
-				return 0;
+	Modules modules;
+	std::string name;
+	std::stringstream stream(maps);
+
+	while (stream.good()) {
+		std::string buf;
+
+		for (auto ch = stream.get(); ch != '-'; ch = stream.get()) {
```

Comment:
```suggestion
		for (char ch = stream.get(); ch != '-'; ch = stream.get()) {
```
why use `auto` here if the actual type has the same amount of characters? ^^

Reply:
Just for consistency.

Also, the function actually returns `int_type`: https://en.cppreference.com/w/cpp/io/basic_istream/get